### PR TITLE
Docs: Encoding & Decoding restructuring

### PR DESCRIPTION
**Pull request by heueristik**
## Description

This PR merges the encoding and decoding sections and introduces more structure to it

Task: [APP-2037](https://aragonassociation.atlassian.net/browse/APP-2037)

## Type of change

- Documentation

[APP-2037]: https://aragonassociation.atlassian.net/browse/APP-2037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/client/examples/04-addresslistVoting-client/01-installation.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* MARKDOWN
```
Comment (brickpop) [inline modules/client/examples/04-addresslistVoting-client/01-installation.ts:1]: No other files changes for Addresslist voting? Token voting and multisig have plenty of renames

File: modules/client/examples/05-encoders-decoders/01-grant-permission.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/* MARKDOWN
```
Comment (brickpop) [inline modules/client/examples/05-encoders-decoders/01-grant-permission.ts]: There's already one file with index `1` right above

File: modules/client/examples/05-encoders-decoders/02-grant-with-condition.ts [hunk 3]
```diff
@@ -15,36 +17,54 @@ import {
...
-// Initializes a general purpose client from the Aragon SDK context.
-const client: Client = new Client(contextPlugin);
+// Instantiates an Aragon OSx SDK client.
+const client: Client = new Client(context);
 
-const grantWithConditionParams: GrantPermissionWithConditionParams = {
+const params: GrantPermissionWithConditionParams = {
   who: "0x1234567890123456789012345678901234567890", // address to which the permission will be granted
   where: "0x2345678901234567890123456789012345678901", // where the permission is granted
   permission: Permissions.EXECUTE_PERMISSION, // the permission to grant. alternatively: UPGRADE_PERMISSION, SET_METADATA_PERMISSION, WITHDRAW_PERMISSION, SET_SIGNATURE_VALIDATOR, SET_TRUSTED_FORWARDER_PERMISSION, ROOT_PERMISSION, CREATE_VERSION_PERMISSION, REGISTER_DAO_PERMISSION, REGISTER_PERMISSION, REGISTER_ENS_SUBDOMAIN_PERMISSION, MINT_PERMISSION, MERKLE_MINT_PERMISSION, MODIFY_ALLOWLIST_PERMISSION, SET_CONFIGURATION_PERMISSION
```
Comment (brickpop) [inline modules/client/examples/05-encoders-decoders/02-grant-with-condition.ts]: WITHDRAW permission, isn't it gone?

File: modules/client/examples/05-encoders-decoders/12-token-voting/03-update-plugin-settings.ts [hunk 4]
```diff
@@ -36,3 +31,35 @@ const pluginAddress: string = "0x1234567890123456789012345678901234567890"; // t
...
+
+/* MARKDOWN
+Returns:
+
+```
+{
+  minDuration: 7200, // seconds
+  minParticipation: 0.25, // 25%
+  supportThreshold: 0.5, // 50%
+  minProposerVotingPower: BigInt("5000")
```
Comment (brickpop) [inline modules/client/examples/05-encoders-decoders/12-token-voting/03-update-plugin-settings.ts]: `BigInt(...)` is used as a "constructor" for legacy environments, but the value you are going to receive would be expressed like `5000n` by a browser

File: modules/client/examples/index.ts [hunk 5]
```diff
@@ -3,9 +3,9 @@
 title: Getting Started
 ---
 
-## Getting Started with the Aragon OSx SDK
+## Getting Started With the Aragon OSx SDK
```
Comment (brickpop) [inline modules/client/examples/index.ts]: The official brand is `Aragon SDK`

```suggestion
## Getting Started with the Aragon SDK
```

File: modules/client/examples/index.ts [hunk 6]
```diff
@@ -3,9 +3,9 @@
...
 
-### Before you begin
+### Before You Begin
```
Comment (brickpop) [inline modules/client/examples/index.ts:8]:
```suggestion
### Before you begin
```

File: modules/client/examples/05-encoders-decoders/01-grant-permission.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+const daoAddress: string = "0x1234567890123456789012345678901234567890";
+
+const action: DaoAction = await client.encoding.grantAction(daoAddress, params);
+console.log({ action });
+
+/* MARKDOWN
+Returns:
+
+```json
+{ grantPermission:
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/01-grant-permission.ts]: If we change the name of the variable from `grantPermission`to action this is no longer correct and must be
```ts
{ action: 
...
}
```

File: modules/client/examples/05-encoders-decoders/04-withdraw.ts [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+  amount: BigInt(10), // the amount in wei to withdraw
+  recipientAddressOrEns: "0x1234567890123456789012345678901234567890" // the address to transfer the funds to
+};
+
+// Withdraws ETH from a given DAO and transfers them to another address.
+const nativeWithdrawAction: DaoAction = await client.encoding.withdrawAction(
+  params
+);
+console.log({ nativeWithdrawAction });
+
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/04-withdraw.ts]: Missing te return of the console log comment

```js
{ nativeWithdrawAction: {
    to: "0x1234567890...",
    value: 10n,
    data: Uint8Array( [0, 10, 20, ...]),
  }
}
```

File: modules/client/examples/05-encoders-decoders/04-withdraw.ts [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+/* MARKDOWN
+Returns:
+
+```json
+{
+  recipientAddress: "0x1234567890123456789012345678901234567890",
+  amount: 10n,
+  tokenAddress: "0x1234567890123456789012345678901234567890",
+  reference: "test"
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/04-withdraw.ts]: Reference was deleted

File: modules/client/examples/05-encoders-decoders/04-withdraw.ts [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+  tokenAddress: "0x1234567890123456789012345678901234567890", // ERC20 token's address
+  amount: BigInt(10), // the amount ignoring the decimals
+  recipientAddressOrEns: "0x1234567890123456789012345678901234567890" // the address to transfer the funds to
+};
+
+const erc20WithdrawAction: DaoAction = await client.encoding.withdrawAction(
+  params
+);
+console.log({ erc20WithdrawAction });
+
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/04-withdraw.ts]: Same here, missing the return

File: modules/client/examples/05-encoders-decoders/04-withdraw.ts [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+/* MARKDOWN
+#### Decoding
+*/
+
+decodedParams = client.decoding.withdrawAction(erc20WithdrawAction.data);
+console.log({ decodedParams });
+
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/04-withdraw.ts]: same

File: modules/client/examples/05-encoders-decoders/05-register-standard-callback.ts [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+const action: DaoAction = client.encoding.registerStandardCallbackAction(
+  dao,
+  params
+);
+console.log({ action });
+
+/* MARKDOWN
+
+```json
+  {
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/05-register-standard-callback.ts:40]: If we log `{ action }`, this is not correct and should be:

```
{ action: {
    to: "0x123123123...",
    value: 0n,
    data: Uint8Array[12,34,45...]
  }
}
```

if we log `action` this is correct

File: modules/client/examples/05-encoders-decoders/05-register-standard-callback.ts [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+/* MARKDOWN
+### Decoding
+*/
+
+const decodedParams: RegisterStandardCallbackParams = client.decoding.registerStandardCallbackAction(action.data);
+console.log({ decodedParams });
+
+/* MARKDOWN
+Returns:
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/05-register-standard-callback.ts:56]: same

File: modules/client/examples/05-encoders-decoders/06-update-metadata.ts [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+// Pins the metadata in IPFS and returns the IPFS URI.
+const ipfsUri: string = await client.methods.pinMetadata(metadataParams);
+
+// Update the metadata of a given DAO.
+const action: DaoAction = await client.encoding.updateDaoMetadataAction(
+  daoAddressOrEns,
+  ipfsUri
+);
+console.log({ action });
+
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/06-update-metadata.ts]: missing return

File: modules/client/examples/05-encoders-decoders/06-update-metadata.ts [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+const decodedParams: DaoMetadata = await client.decoding.updateDaoMetadataAction(
+  action.data
+);
+console.log({ decodedParams });
+
+/* MARKDOWN
+Returns:
+
+```json
+{
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/06-update-metadata.ts]: the return is not correct as pointed out before

File: modules/client/examples/05-encoders-decoders/07-set-dao-uri.ts [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+  daoUri
+);
+console.log({ action });
+
+/* MARKDOWN
+Returns:
+
+```json
+  {
+    to: "0x123123123...",
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/07-set-dao-uri.ts]: same again, maybe fixing this in a new PR or task?

File: modules/client/examples/05-encoders-decoders/12-multisig/01-add-addresses.ts [hunk 17]
```diff
@@ -6,20 +6,15 @@ title: Add Members
 ## Add Members from the Multisig Plugin
 
 Adds new address as members of the Multisig plugin installed in a DAO, so they are now able to vote on proposals.
+
```
Comment (josemarinas) [inline modules/client/examples/05-encoders-decoders/12-multisig/01-add-addresses.ts:9]: We dont have a findInterface example for multisig but we do have it for the other plugins

File: modules/client/examples/04-addresslistVoting-client/01-installation.ts (hunk 1 above)
Comment (heueristik) [reply modules/client/examples/04-addresslistVoting-client/01-installation.ts:1]: Yes. For  TokenVoting and Multisig, the parent folders got renamed

File: modules/client/examples/05-encoders-decoders/01-grant-permission.ts (hunk 2 above)
Comment (heueristik) [reply modules/client/examples/05-encoders-decoders/01-grant-permission.ts]: Fixed in 2f284e5bbe93a665e8ffd13477ee3fc44d81b3be

File: modules/client/examples/index.ts (hunk 6 above)
Comment (heueristik) [reply modules/client/examples/index.ts:8]: I followed https://capitalizemytitle.com/, also for the osx docs

File: modules/client/examples/index.ts (hunk 5 above)
Comment (heueristik) [reply modules/client/examples/index.ts]: Fixed in commit 6fc88cb2d013aac7f2c766765eb305afee585ffc and used https://capitalizemytitle.com/ for capitalization

File: modules/client/examples/05-encoders-decoders/12-token-voting/03-update-plugin-settings.ts (hunk 4 above)
Comment (heueristik) [reply modules/client/examples/05-encoders-decoders/12-token-voting/03-update-plugin-settings.ts]: Yes, the comments also won't be returned and so on. I just copied it over. Someone should go over all the example and polish this

File: modules/client/examples/05-encoders-decoders/02-grant-with-condition.ts (hunk 3 above)
Comment (heueristik) [reply modules/client/examples/05-encoders-decoders/02-grant-with-condition.ts]: Yes. Moreover, this is not the right place to provide and exhaustive list of all permissions but just an example. So I removed the list in 37d7db46960470bbb0946c48ba96865a86dabab1

File: modules/client/examples/05-encoders-decoders/12-token-voting/03-update-plugin-settings.ts (hunk 4 above)
Comment (heueristik) [reply modules/client/examples/05-encoders-decoders/12-token-voting/03-update-plugin-settings.ts]: @josemarinas will take care about this in the next iteration

**Review by josemarinas — APPROVED** [review]

**Pull request merged**